### PR TITLE
#2227: Add Sort to search page

### LINKED ISSUE #2227: [feature]: Add Sorting feature to Search Results Page
**Is your feature request related to a problem? Please describe.**
Category sorting was implemented in #360 and now needs to also be implemented for the search results page. 

As a shopper, after I use the search feature to find specific items in the store I want to be able to sort the returned products by the available criteria: price (low to high), price (high to low), best match

**Describe the solution you'd like**
Implement existing category sorting feature on search results page. 

Sort button should display and support all currently available sorting options

**Describe alternatives you've considered**

**Please let us know what packages this feature is in regards to:**
- [ ] `venia-concept`
- [x ] `venia-ui`
- [ ] `pwa-buildpack`
- [ x] `peregrine`
- [ ] `pwa-devdocs`
- [ ] `upward-js`
- [ ] `upward-spec`
- [ ] `create-pwa`

### PR DESCRIPTION
**Pull request by larsroettig**
## Description

Add Sorting feature to Search Results Page

## Related Issue

Closes #2227 

### Verification Steps

1. search for `tops`
2. sort is visible on search page

## Checklist

* I have updated the documentation accordingly, if necessary.
* I have added tests to cover my changes, if necessary.

### CODE REVIEW

File: packages/venia-ui/lib/components/SearchPage/searchPage.js [hunk 1]
```diff
@@ -12,16 +12,30 @@ import defaultClasses from './searchPage.css';
...
+        sortAttribute: 'relevance',
+        sortDirection: 'ASC'
+    });
+
+    const { sortAttribute, sortDirection } = sort;
+
+    const sortControl = {
+        currentSort: sort,
+        setSort: setSort
+    };
```
Comment (tjwiebell) [inline packages/venia-ui/lib/components/SearchPage/searchPage.js:30]: All of this logic, and the label logic below should all be contained within the `useSearchPage` talon. That should make it a little cleaner when determining the label; you could use two Maps and concat labels, or a single Map with hashed keys that are a combination of attribute/direction.

File: packages/peregrine/lib/talons/SearchPage/useSearchPage.js [hunk 2]
```diff
@@ -113,7 +116,8 @@ export const useSearchPage = props => {
                 currentPage: Number(currentPage),
                 filters: newFilters,
                 inputText,
-                pageSize: Number(PAGE_SIZE)
+                pageSize: Number(PAGE_SIZE),
+                sort: { [String(sortAttribute)]: String(sortDirection) }
```
Comment (tjwiebell) [inline packages/peregrine/lib/talons/SearchPage/useSearchPage.js:120]: I see that you did this in the category implementation, but casting is usually a smell, unless that data type is out of your control (ie. data fetch). If you can guarantee this state will always be a string, I would consider removing this.

**Comment by tjwiebell** [comment]
@larsroettig - Just checking in to see how this is going. All the logic is there, so we're happy to pull this internal to finish up if your availability is light. Please let me know if there's anything we can do to help.

**Comment by larsroettig** [comment]
@tjwiebell will finish today late night should be completed before you start working tomorrow

File: packages/venia-ui/lib/components/SearchPage/searchPage.css [hunk 3]
```diff
@@ -23,12 +23,21 @@
...
     border: 1px solid black;
-    border-radius: 100px;
+    border-radius: 3px;
     color: black;
     outline: none;
 }
+
+.sortContainer {
+    font-size: 0.875rem;
+    font-style: italic;
```
Comment (jimbo) [inline packages/venia-ui/lib/components/SearchPage/searchPage.css:37]: Avoid using `italic`. We don't load that font style (or want to), so the browser has to emulate it, which is no good. To reduce the prominence of text, use a lighter color or decrease the size.

**Comment by larsroettig** [comment]
@tjwiebell as we speak the new issues to refactor it to hook:
https://github.com/magento/pwa-studio/issues/2331

File: packages/venia-ui/lib/components/SearchPage/searchPage.css (hunk 3 above)
Comment (tjwiebell) [reply packages/venia-ui/lib/components/SearchPage/searchPage.css]: Blindly followed the mock, good to know. Removed in 5cf9e9a.

**Pull request merged**